sg_otio/track_diff.py

```diff
@@ -603,6 +683,70 @@ def old_clip_for_shot(cls, for_clip, prev_clip_list, sg_shot, sg_version=None):
             return best
         return None
 
+    def get_cut_diff_for_clip(self, clip, index, sg_shot=None, as_omitted=False, repeated=False):
+        """
+        Return a new :class:`SGCutDiff` for the given clip.
+
+        This can be re-implemented in deriving classes to return a custom instance
+        deriving from a :class:`SGCutDiff`.
+
+        :param clip: A :class:`otio.schema.Clip` instance.
+        :param int index: The index of the clip in its track.
+        :param sg_shot: A SG Shot dictionary.
+        :param bool as_omitted: Whether this is a difference for an omitted Shot or not.
+        :param bool repeated: Whether this is a difference for a Shot with multiple edits.
+        """
+        return SGCutDiff(
+            clip=clip,
+            index=index,
+            sg_shot=sg_shot,
+            as_omitted=as_omitted,
+            repeated=repeated,
+        )
+
+    def add_cut_diff(self, shot_name, clip, index, sg_shot=None, as_omitted=False):
+        """
+        Add a new Cut difference to this :class:`TrackDiff`.
+
+        :param shot_name: Shot name, as a string.
+        :param clip: A :class:`otio.schema.Clip` instance.
+        :param int index: The index of the clip in its track.
+        :param sg_shot: A SG Shot dictionary.
+        :param bool as_omitted: Whether this is a difference for an omitted Shot or not.
+        """
+        # Force a lowercase key to make Shot names case-insensitive. Shot names
+        # we retrieve from editorial files may be uppercase, but actual SG Shots may be
+        # lowercase.
+        # We might not have a valid Shot name if we have an edit without any
+        # Shot name or Version name. To avoid considering all these entries
+        # as repeated Shots we forge a key based on the cut order.
+        shot_key = shot_name.lower() if shot_name else shot_name  # else "_no_shot_name_%s" % index
```

Comment:
is the comment needed?

Reply:
yes, but needs to be improved to explain what was done in the old implementation and could be needed back (not sure).